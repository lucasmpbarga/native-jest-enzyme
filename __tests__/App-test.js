import React from 'react';
import {mount, shallow} from 'enzyme';
import App, {renderFlatList} from '../App';

describe('Button', () => {
  describe('Rendering', () => {
    const setupRenderFlatList = data => mount(renderFlatList(data));

    it('should match to snapshot', () => {
      const wrapper = mount(<App />);
      expect(wrapper).toMatchSnapshot();
    });

    it('should match to snapshot 2', () => {
      const wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot();
    });

    it('new test', () => {
      const mockData = [
        {
          id: 'bd7acbea-c1b1-46c2-aed5-3ad53abb28ba',
          title: 'First Item',
        },
        {
          id: '3ac68afc-c605-48d3-a4f8-fbd91aa97f63',
          title: 'Second Item',
        },
        {
          id: '58694a0f-3da1-471f-bd96-145571e29d72',
          title: 'Third Item',
        },
      ];

      const wrapper = setupRenderFlatList(mockData);

      expect(wrapper.text()).toContain('First Item');
      expect(wrapper.text()).toContain('Second Item');
      expect(wrapper.text()).toContain('Third Item');
    });
  });
});
