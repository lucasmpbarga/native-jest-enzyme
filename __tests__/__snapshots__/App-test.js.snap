// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Rendering should match to snapshot 1`] = `ReactWrapper {}`;

exports[`Button Rendering should match to snapshot 2 1`] = `ShallowWrapper {}`;
